gazebo_ros/test/test_disable_performance_metrics.test.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2021 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import launch
+import launch.actions
+
+import launch_testing
+import launch_testing.asserts
+import pytest
+import os
+
+import rclpy
+from gazebo_msgs.msg import PerformanceMetrics
+from rclpy.node import Node
+from rcl_interfaces.msg import Parameter, ParameterType
+from rcl_interfaces.srv import SetParameters
+import time
+from threading import Thread, Event
+
+# Global varibales to track interval between messages
+msg_interval = None
+start_time = None
+
+
+@pytest.mark.launch_test
+def generate_test_description():
+    """Launch a gzserver node."""
+    return launch.LaunchDescription([
+        launch.actions.ExecuteProcess(
+            cmd=['gzserver', '-s', 'libgazebo_ros_init.so'],
+            output='screen',
+            sigterm_timeout='20',
+            sigkill_timeout='20'
+        ),
+        launch_testing.actions.ReadyToTest()
+    ])
+
+
+class TestPerformanceMetricsParam_1(unittest.TestCase):
+    """Test to set parameter to true."""
+
+    @classmethod
+    def setUpClass(cls):
+        rclpy.init()
+
+    @classmethod
+    def tearDownClass(cls):
+        rclpy.shutdown()
+
+    def test_parameter_enable(self):
+        """
+        Test function to set parameter to true.
+
+        Checks if /performance_metrics topic starts publishing if
+        enable_performance_metrics is set to True.
+        """
+        node = MakeTestNode('performance_metrics_enable_testing_node')
+
+        assert node.wait_for_gazebo_node(timeout_sec=5.0),\
+            'Gazebo node not found, parameter could not be set'
+
+        # Set enable_performance_metrics parameter to True and listen for messages
+        response = node.set_parameter(state=True)
+        assert response.successful, 'Parameter could not be set to True'
+
+        node.start_subscriber()
+        msgs_received_flag = node.msg_event_object.wait(timeout=5.0)
+        global msg_interval
+        assert msgs_received_flag, 'Did not receive any messages after\
+            setting enable_performance_metrics to True, test failed'
+
+
+class TestPerformanceMetricsParam_2(unittest.TestCase):
```

Comment:
I don't think we need a separate test fixture for the second test; we can keep both tests associated with the same fixture,  ie.

```py
class TestPerformanceMetricsParam(unittest.TestCase):
    
    @classmethod
    def setUpClass(cls):
        rclpy.init()

    @classmethod
    def tearDownClass(cls):
        rclpy.shutdown()

    def test_parameter_enable(self):
        ...
    
    def test_parameter_disable(self):
        ...
```

Reply:
Done. However, ```rclpy.shutdown()`` has to be called after the end of both of them to stop the spin thread, so that the next node can spin.